### DISCUSSION TITLE
How  to generate listing items report with the latest updated products in a datetime range?

### DISCUSSION THREAD

**Discussion by brunoamorimdev** [amzn/selling-partner-api-models#4151] [Q&A] [answered]
Hi,

I am seeking a solution to retrieve all listed items that have been updated within a specific date or datetime range. If it’s not feasible to directly retrieve a list of items updated in that time frame, is there a report available that includes a column showing the latest update date or datetime for each item?

Any pointers to the documentation would be great!

Bruno,

**Comment by mafge** [comment]
Hi @brunoamorimdev,
thanks for reaching out!
Unfortunately there is no such report. The only way to retrieve last update data for a listing is using [getListingsItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#getlistingsitem) which must be called separately for each listing item. Hope that helps!
Best,
Marc
Selling Partner Developer Services

**Reply by brunoamorimdev** [reply]
Hi @mafge,

Thanks for the quick response.

I realize I may not have explained my use case clearly. What I need is to retrieve only the listing items that were updated within a specific date range. The simplest approach, as I see it, would be to compare the current date and time with the most recent update timestamp for each listing item.

Initially, I thought I could get this information through a notification, but it appears no such notification is available at the moment. An alternative approach I considered was requesting a report on listing items that includes this data. However, the reports I’ve reviewed so far do not provide this field.

It seems the only remaining option is to track changes manually by requesting the seller’s product report, iterating through each SKU, and checking for any differences from the previously saved data. I’m wondering if the SP API offers any feature that could streamline this process of checking for changes in listing items.

Best regards,
Bruno

**Reply by mafge** [reply, marked as answer]
Hi @brunoamorimdev,
your assumptions so far are correct. There might be other options if you want to share your use case with me, but with the information provided there is no other option than tracking changes manually.

**Reply by brunoamorimdev** [reply]
Hi @mafge,

Yeah, I guess that's it. 
Thanks for the support.

Best regards,
Bruno
